.github/workflows/end2end-test.yml

```diff
@@ -127,7 +131,7 @@ jobs:
               # Don't fail the job if there isn't any flaky tests report.
               continue-on-error: true
               with:
-                  name: flaky-tests-report
+                  name: flaky-tests-report-${{ matrix.part }}
```

Comment:
Seems like we can use the [`pattern`](https://github.com/actions/upload-artifact/blob/main/docs/MIGRATION.md) input to download all artifacts in one step rather than splitting them into multiple parts.